### PR TITLE
Add costing model for coagulation/flocculation unit

### PR DESCRIPTION
**Pull request by luohezhiming**
## Summary/Motivation:

Add costing model for coagulation/flocculation

## Changes proposed in this PR:
- add costing model
- modify `total_power` to `electricity` and register electricity
- add corresponding tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/core/zero_order_costing.py [hunk 1]
```diff
@@ -518,6 +519,84 @@ def cost_chlorination(blk):
...
+                cost_floc,
+                to_units=blk.config.flowsheet_costing_block.base_currency) +
+            pyo.units.convert(
+                cost_coag_inj,
+                to_units=blk.config.flowsheet_costing_block.base_currency) +
+            pyo.units.convert(
+                cost_floc_inj,
+                to_units=blk.config.flowsheet_costing_block.base_currency))
+
+        expr *= blk.config.flowsheet_costing_block.TPEC
```
Comment (andrewlee94) [inline watertap/core/zero_order_costing.py]: Whilst the original WT3 models hard-coded this decision, I have generally been setting this in the YAML file so as to allow more flexibility in future.

File: watertap/unit_models/zero_order/tests/test_coag_and_floc_zo.py [hunk 2]
```diff
@@ -17,17 +17,20 @@
 
 from io import StringIO
 from pyomo.environ import (
+    Block, check_optimal_termination,
```
Comment (andrewlee94) [inline watertap/unit_models/zero_order/tests/test_coag_and_floc_zo.py]: Is `check_optimal_termination` being used in this test? I did a quick search and couldn't see it being used, which would be on oversight, as we should be testing that we can solve the model.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
Two fairly minor things that should be fixed, but otherwise looks good.

File: watertap/unit_models/zero_order/tests/test_coag_and_floc_zo.py (hunk 2 above)
Comment (luohezhiming) [reply watertap/unit_models/zero_order/tests/test_coag_and_floc_zo.py]: No, this is not used in the test. I cleaned up this.

File: watertap/core/zero_order_costing.py (hunk 1 above)
Comment (luohezhiming) [reply watertap/core/zero_order_costing.py]: OK. I revised following the convention and added an option for choosing from `TPEC` or `TIC`.

File: watertap/unit_models/zero_order/tests/test_coag_and_floc_zo.py (hunk 2 above)
Comment (adam-a-a) [reply watertap/unit_models/zero_order/tests/test_coag_and_floc_zo.py]: `assert_optimal_termination` was (and still is) included to account for this.

**Review by andrewlee94 — APPROVED** [review]

**Review by adam-a-a — APPROVED** [review]

**Pull request merged**